crates/turbo-tasks-memory/src/task.rs

```diff
@@ -620,16 +692,71 @@ impl Task {
         schedule_task
     }
 
+    /// When any scope is active it returns true. When no scope is active it
+    /// returns false and adds the tasks to all scopes as dirty task.
+    /// When `increment_unfinished` is true it will also increment the
+    /// unfinished tasks for all scopes, independent of activeness.
+    fn scopes_dirty_or_active(
+        &self,
+        increment_unfinished: bool,
+        scopes: &TaskScopes,
+        backend: &MemoryBackend,
+    ) -> bool {
+        if increment_unfinished {
+            // We need to walk all scopes at least once to increment unfinished tasks.
+            // While doing that we check if any scope is active.
+            let mut active = false;
+            for scope in scopes.iter() {
+                backend.with_scope(scope, |scope| {
+                    scope.increment_unfinished_tasks(backend);
+                    active = active || scope.state.lock().is_active();
+                })
+            }
+            if active {
+                return true;
+            }
+        } else {
+            // Without the need to increment unfinished for all scopes we can exit early
+            if scopes
+                .iter()
+                .any(|scope| backend.with_scope(scope, |scope| scope.state.lock().is_active()))
+            {
+                return true;
+            }
+        }
+        for (i, scope) in scopes.iter().enumerate() {
+            if backend.with_scope(scope, |scope| {
+                let mut state = scope.state.lock();
+                let is_active = state.is_active();
+                if !is_active {
```

Comment:
Is it actually possible to find an active scope? In both branches at the start of the function, if any active scope is found, we've already exited the function.